### PR TITLE
Enable editing and delete info comments patch

### PR DESCRIPTION
**Pull request by pontus-osterdahl**
This pull request adds the functionality to edit and delete info-comments. Fixes https://github.com/kitodo/kitodo-production/issues/5702 . Fixes issues with pull request #5703 

* Add editCommentDialog.xhtml
* Enable removal of Comment in CommentService and CommentsForm
* Add integration test for CommentService

 * Changed properties of comments association in Process bean

* Add messages for editing and deleting comments
* Update kitodo.css

### CODE REVIEW

**Comment by solth** [comment]
@pontus-osterdahl thanks for opening a new pull request! This looks much better now! Unfortunately, merging the older PR #5657 now created a conflict with your code. Could you please resolve the conflict? Then this PR is next to merge candidate.

**Review by solth — COMMENTED** [review]
@pontus-osterdahl Thank you for resolving the conflict!

I just tested your branch and it worked as intended. I just noticed some small issue: when deleting the last comment of a process, the "comments" column in the process list still shows the icon to mark the existence of comments, but the mouse over popup is empty:

Bildschirmfoto 2023-10-06 um 14 01 59

It seems this happens because the `ProcessDTO.hasComments` property is not re-evaluated when deleting existing comments.

**Comment by pontus-osterdahl** [comment]
Thanks @solth for noticing this issue. The process containing the edited/deleted comment was not indexed. I have pushed a patch which should solve the issue. Or should we await the Hibernate Search integration?

**Review by solth — COMMENTED** [review]
Seems to work very well now. I have just one more minor change suggestion to slightly improve the code at one place (see above).

File: Kitodo/src/main/java/org/kitodo/production/forms/CommentForm.java [hunk 1]
```diff
@@ -69,7 +69,8 @@ public Process getProcess() {
     public void removeComment(Comment comment) {
         try {
             ServiceManager.getCommentService().removeComment(comment);
```
[the remark this answers is not in the document]
Comment (pontus-osterdahl) [reply Kitodo/src/main/java/org/kitodo/production/forms/CommentForm.java:71]: Thanks again! I agree and would tend to make it non static.

File: Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java [hunk 2]
```diff
@@ -11,15 +11,19 @@
...
 
+import java.io.IOException;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 
 import org.kitodo.data.database.beans.Comment;
 import org.kitodo.data.database.beans.Process;
 import org.kitodo.data.database.beans.Task;
+import org.kitodo.data.database.beans.User;
```
Comment (solth) [inline Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java:22]:
```suggestion
```
Unused import

File: Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java [hunk 3]
```diff
@@ -11,15 +11,19 @@
 
 package org.kitodo.production.services.data;
 
+import java.io.IOException;
```
Comment (solth) [inline Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java:14]:
```suggestion
```
Unused import

File: Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java [hunk 4]
```diff
@@ -11,15 +11,19 @@
...
 import java.util.Map;
 import java.util.Objects;
 
 import org.kitodo.data.database.beans.Comment;
 import org.kitodo.data.database.beans.Process;
 import org.kitodo.data.database.beans.Task;
+import org.kitodo.data.database.beans.User;
 import org.kitodo.data.database.exceptions.DAOException;
 import org.kitodo.data.database.persistence.CommentDAO;
+import org.kitodo.data.exceptions.DataException;
```
Comment (solth) [inline Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java:25]:
```suggestion
```
Unused import

File: Kitodo/src/main/webapp/WEB-INF/templates/includes/comments/editCommentDialog.xhtml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ *
+-->
+
+<ui:composition
+        xmlns="http://www.w3.org/1999/xhtml"
+        xmlns:h="http://xmlns.jcp.org/jsf/html"
+        xmlns:f="http://xmlns.jcp.org/jsf/core"
```
Comment (solth) [inline Kitodo/src/main/webapp/WEB-INF/templates/includes/comments/editCommentDialog.xhtml:17]:
```suggestion
```
Unused xml namespace

File: Kitodo/src/test/java/org/kitodo/production/services/data/CommentServiceIT.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+package org.kitodo.production.services.data;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Date;
+import java.util.List;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
```
Comment (solth) [inline Kitodo/src/test/java/org/kitodo/production/services/data/CommentServiceIT.java:21]:
```suggestion
```
Unused import

File: Kitodo/src/test/java/org/kitodo/production/services/data/CommentServiceIT.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
```
Comment (solth) [inline Kitodo/src/test/java/org/kitodo/production/services/data/CommentServiceIT.java:23]:
```suggestion
```
Unused import

File: Kitodo/src/test/java/org/kitodo/production/services/data/CommentServiceIT.java [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+import org.kitodo.MockDatabase;
+import org.kitodo.data.database.beans.Comment;
+import org.kitodo.data.database.beans.Process;
+import org.kitodo.data.database.beans.Task;
```
Comment (solth) [inline Kitodo/src/test/java/org/kitodo/production/services/data/CommentServiceIT.java:27]:
```suggestion
```
Unused import

**Review by solth — CHANGES_REQUESTED** [review]
Thanks again. Just found a few unused import that should be removed before merging.

**Review by solth — APPROVED** [review]

**Pull request merged**